node/flow.js

```diff
@@ -0,0 +1,39 @@
+module.exports = {
+  parser: 'babel-eslint',
+  extends: 'mixmax/node',
+  plugins: [
+    // Allows correct parsing of flow annotated files.
+    'flowtype'
+  ],
+  rules: {
+    'space-infix-ops': 0,
+    'no-duplicate-imports': 0,
+    // Makes flow `type` definitions defines
+    'flowtype/define-flow-type': 1,
+    // Do not allow spaces before type declarations, i.e. let a: number, not let a : number
+    'flowtype/space-before-type-colon': [
+      1,
+      'never'
+    ],
+    // Marks flow type declarations as used to prevent no-unused-vars for types
+    'flowtype/use-flow-type': 1,
+    // Enforces camelcase for type definitions
+    'flowtype/type-id-match': [
+      2,
+      '^([A-Z]+[a-z0-9A-Z]*)$'
+    ]
+  },
+  settings: {
+    // Only use the flow linter for files with `@flow`
+    flowtype: {
+      onlyFilesWithFlowAnnotation: true
+    }
+  },
+  parserOptions: {
+    // Required to support import/export syntax when using types
+    sourceType: 'module',
```

Comment:
what if we want to use this server-side with `script`-type `sourceType`?

i'm wondering if we want a base `flow` config that could be mixed with other configs to make eg `flow-node` and `flow-browser`.

Reply:
To the first point, this works with server-side code just fine as far as I can tell (https://github.com/mixmaxhq/event-logging/pull/1 this correctly enforces rules using this config). On the other point, we could not extend `mixmax/node` here and instead rely on the client to extend it like `"extends": [ "mixmax/node", "mixmax/flow" ]`